tests/tests/test-staking/test-weights.ts

```diff
@@ -0,0 +1,322 @@
+          .signAsync(account)
+      )
+    )
+  );
+});
+
+describeBenchmark("Staking - Max Transaction Fit", "scheduleLeaveDelegators", async (context) => {
+  const maxTransactions = 350;
+  const randomAccounts = await createAccounts(context, maxTransactions);
+  for (const randomAccountsChunk of chunk(randomAccounts, 20)) {
+    await expectOk(
+      context.createBlock(
+        randomAccountsChunk.map((account) =>
+          context.polkadotApi.tx.parachainStaking
+            .delegateWithAutoCompound(
+              alith.address,
+              MIN_GLMR_DELEGATOR,
+              100,
+              maxTransactions,
+              maxTransactions,
+              0
+            )
+            .signAsync(account)
+        )
+      )
+    );
+  }
+
+  expect((await context.polkadotApi.query.parachainStaking.delegatorState.keys()).length).to.equal(
+    maxTransactions
+  );
+
+  await expectOk(
+    context.createBlock(
+      randomAccounts.map((account) =>
+        context.polkadotApi.tx.parachainStaking.scheduleLeaveDelegators().signAsync(account)
+      )
+    )
+  );
+});
+
+describeBenchmark("Staking - Max Transaction Fit", "executeLeaveDelegators", async (context) => {
+  const maxTransactions = 350;
+  const randomAccounts = await createAccounts(context, maxTransactions);
+
+  await expectOk(
+    context.createBlock(
+      context.polkadotApi.tx.sudo
+        .sudo(context.polkadotApi.tx.parachainStaking.setBlocksPerRound(10))
+        .signAsync(alith)
+    )
+  );
+
+  for (const randomAccountsChunk of chunk(randomAccounts, 20)) {
+    await expectOk(
+      context.createBlock(
+        randomAccountsChunk.map((account) =>
+          context.polkadotApi.tx.parachainStaking
+            .delegateWithAutoCompound(
+              alith.address,
+              MIN_GLMR_DELEGATOR,
+              100,
+              maxTransactions,
+              maxTransactions,
+              0
+            )
+            .signAsync(account)
+        )
+      )
+    );
+  }
+
+  expect((await context.polkadotApi.query.parachainStaking.delegatorState.keys()).length).to.equal(
+    maxTransactions
+  );
+
+  for (const randomAccountsChunk of chunk(randomAccounts, 20)) {
+    await expectOk(
+      context.createBlock(
+        randomAccountsChunk.map((account) =>
+          context.polkadotApi.tx.parachainStaking.scheduleLeaveDelegators().signAsync(account)
+        )
+      )
+    );
+  }
+
+  await jumpRounds(context, 3);
+
+  await expectOk(
+    context.createBlock(
+      randomAccounts.map((account) =>
+        context.polkadotApi.tx.parachainStaking
+          .executeLeaveDelegators(account.address, 1)
+          .signAsync(account)
+      )
+    )
+  );
+});
+
+// utils
+
+function describeBenchmark(
+  title: string,
+  method: string,
+  cb: (context: DevTestContext) => Promise<void>
+) {
+  describeDevMoonbeam(`${title} - ${method}`, (context) => {
+    it("should fit minimum 2", async function () {
```

Comment:
We could parameterize this (2) and make these a permanent test. It might add a lot of time to the tests, though.

Reply:
There's a conditional env variable `STAKING_BENCHMARK=1` that is checked whether or not to run this test, but I'm all up for parameterizing if it adds value.